core/eolearn/core/eodata.py

```diff
@@ -78,7 +79,11 @@ def __init__(self, feature_dict: Mapping[str, T | FeatureIO[T]], feature_type: F
         super().__init__()
 
         self.feature_type = feature_type
-        self._content = dict(feature_dict.items())
+
+        # we need trigger parsing and validation
+        self._content: dict[str, T | FeatureIO[T]] = {}
+        for key, value in feature_dict.items():
+            self[key] = value
```

Comment:
Not sure I understand this

Reply:
`self._content` is just a regular dictionary, no checks are done on it.
So instead of directly filling the `_content` we feed the values through `self[key]` so that our custom `__setitem__` triggers and parses/validates the values. Before you could sneak in anything at init, now you cannot (I added tests to make sure of that)